### ISSUE TITLE
Test failure: Expect100Continue_WaitsExpectedPeriodOfTimeBeforeSendingContent

### ISSUE THREAD

**Issue by karelz** [dotnet/runtime#58628]
Test types:
- `System.Net.Http.Functional.Tests.SocketsHttpHandler_HttpClientHandler_Cancellation_Test_Http2`
- `System.Net.Http.Functional.Tests.SyncHttpHandler_HttpClientHandler_Cancellation_Test`

Failures 3/12-9/3 (incl. PRs):

Day | Run | OS | Test
-- | -- | -- | --
8/25 | Official run - release/6.0-rc1 | SLES.12.Amd64.Open | SocketsHttpHandler_HttpClientHandler_Cancellation_Test_Http2
8/27 | PR #58092 (main = 7.0) | Ubuntu.1804.Amd64.Open | SyncHttpHandler_HttpClientHandler_Cancellation_Test
9/2 | PR #58336 (release/6.0) | Ubuntu.1604.Amd64.Open | SyncHttpHandler_HttpClientHandler_Cancellation_Test

Failure:
```
System.TimeoutException : The operation has timed out.

   at System.Net.Http.Functional.Tests.SocketsHttpHandler_Cancellation_Test.Expect100Continue_WaitsExpectedPeriodOfTimeBeforeSendingContent() in /_/src/libraries/System.Net.Http/tests/FunctionalTests/SocketsHttpHandlerTest.Cancellation.cs:line 111
--- End of stack trace from previous location ---
```

```json
{
  "ErrorMessage": "Expect100Continue_WaitsExpectedPeriodOfTimeBeforeSendingContent [FAIL]",
  "BuildRetry": false,
  "ExcludeConsoleLog": false
}
```

## Build Information
Build: https://dev.azure.com/dnceng-public/public/_build/results?buildId=438937
Build error leg or test failing: Expect100Continue_WaitsExpectedPeriodOfTimeBeforeSendingContent

 ### Known issue validation
**Build: :mag_right:** https://dev.azure.com/dnceng-public/public/_build/results?buildId=438937
**Error message validated:** `Expect100Continue_WaitsExpectedPeriodOfTimeBeforeSendingContent [FAIL]`
**Result validation: :white_check_mark:** Known issue matched with the provided build.
**Validation performed at:** 10/17/2023 7:26:38 AM UTC

### Report

|Build|Definition|Test|Pull Request|
|---|---|---|---|
|[705030](https://dev.azure.com/dnceng-public/public/_build/results?buildId=705030)|dotnet/runtime|[System.Net.Http.Functional.Tests.SyncHttpHandler_HttpClientHandler_Cancellation_Test.Expect100Continue_WaitsExpectedPeriodOfTimeBeforeSendingContent](https://dev.azure.com/dnceng-public/public/_build/results?buildId=705030&view=ms.vss-test-web.build-test-results-tab&runId=17582014&resultId=104396)||
|[698071](https://dev.azure.com/dnceng-public/public/_build/results?buildId=698071)|dotnet/runtime|[System.Net.Http.Functional.Tests.SocketsHttpHandler_HttpClientHandler_Http11_Cancellation_Test.ConnectTimeout_TimesOutSSLAuth_Throws](https://dev.azure.com/dnceng-public/public/_build/results?buildId=698071&view=ms.vss-test-web.build-test-results-tab&runId=17366998&resultId=103950)||
#### Summary
|24-Hour Hit Count|7-Day Hit Count|1-Month Count|
|---|---|---|
|0|0|2|

**Comment by carlossanlop** [comment]
This issue seems to be a bit old but the failure message is the same I'm seeing every now and then.

This release/7.0-rc2 PR hit it: https://github.com/dotnet/runtime/pull/75932

- Queue: Libraries Test Run release coreclr Linux x64 Debug
- Job results: https://dev.azure.com/dnceng-public/public/_build/results?buildId=24574&view=logs&j=71afe768-e493-597c-228c-037834193aeb&t=fe87d331-02fa-5ca6-e382-69b0a11bb0f6&l=104

```
System.Net.Http.Functional.Tests.SyncHttpHandler_HttpClientHandler_Cancellation_Test.GetAsync_CancelDuringResponseBodyReceived_Unbuffered_TaskCanceledQuickly(chunkedTransfer: True, connectionClose: False, readOrCopyToAsync: False) [FAIL]
      System.TimeoutException : The operation has timed out.
      Stack Trace:
        /_/src/libraries/Common/tests/System/Net/Http/HttpClientHandlerTest.Cancellation.cs(260,0): at System.Net.Http.Functional.Tests.HttpClientHandler_Cancellation_Test.<>c__DisplayClass4_2.<<GetAsync_CancelDuringResponseBodyReceived_Unbuffered_TaskCanceledQuickly>b__0>d.MoveNext()
        --- End of stack trace from previous location ---
        /_/src/libraries/Common/tests/System/Net/Http/LoopbackServer.cs(101,0): at System.Net.Test.Common.LoopbackServer.CreateServerAsync(Func`2 funcAsync, Options options)
        /_/src/libraries/Common/tests/System/Net/Http/HttpClientHandlerTest.Cancellation.cs(228,0): at System.Net.Http.Functional.Tests.HttpClientHandler_Cancellation_Test.GetAsync_CancelDuringResponseBodyReceived_Unbuffered_TaskCanceledQuickly(Boolean chunkedTransfer, Boolean connectionClose, Boolean readOrCopyToAsync)
        --- End of stack trace from previous location ---
```

**Comment by liveans** [comment]
This should be fixed by #102699. I don't see any `TimeoutException` on Kusto within last 30 day as well (only handshake timeouts on QUIC which can be too short in some cases), and both reported issues are from `release/6.0` and `release/8.0` branches. I'd say let's close this and track this handshake timeout issue under another issue.

/cc @ManickaP

**Comment by liveans** [comment]
Original bug fixed by #102699 in main. 
For handshake timeouts closing this in favor of #104426.
